wc-src/vf-voice.js

```diff
@@ -0,0 +1,83 @@
+import Vex from '../src/index.js';
+
+const template = document.createElement('template');
+template.innerHTML = `
+`;
+
+function concat(a, b) {
+  return a.concat(b);
+}
+
+export class VFVoice extends HTMLElement {
+  constructor() {
+    super();
+
+    this.attachShadow({ mode:'open' });
+    this.shadowRoot.appendChild(document.importNode(template.content, true));
+
+    // Defaults
+    this.stem = 'up';
+    this.autoBeam = false;
+    this.notes = [];
+    this.beams = [];
+
+    this._vf = undefined;
+    this._score = undefined;
+  }
+
+  connectedCallback() {
+    this.stem = this.getAttribute('stem') || this.stem;
+    this.autoBeam = this.hasAttribute('autoBeam');
+    this.notesText = this.textContent.trim();
+
+    const getFactoryEvent = new CustomEvent('getFactory', { bubbles: true });
+    this.dispatchEvent(getFactoryEvent);
+
+    const getScoreEvent = new CustomEvent('getScore', { bubbles: true });
+    this.dispatchEvent(getScoreEvent);
```

Comment:
I would suggest emitting a single event, such as 'VFComponentReady' or something. Then vf-score could listen to that and know to set both factory and score.

Reply:
The score is actually handled by the vf-stave, so a single event can't capture both of them. I don't know how this would work with the timing/order of everything, but if vf-stave gets its factory set with a getFactory event to vf-score, could vf-voice dispatch a single event to vf-stave to get both?

Reply:
Ah, yes I think a single event would still work great. vf-voice could dispatch 'VoiceReady' or something, and both vf-stave and vf-score could listen for that and set their respective objects. We just won't be able to get rid of the `(this._vf && this._score)` condition in `createNotes`, but that's fine.